### PR TITLE
Task 6 - PATCH /api/articles/:article_id

### PR DESCRIPTION
**Pull request by clam119**
For this PR I have added the following:

1.  Error.js - Custom file to store the error handling function middleware.
2. App.js - Added the PATCH method to the endpoint of /api/articles/:article_id 
3. App.test.js - Added the tests for status codes of 200, 400 and 404.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -99,3 +99,63 @@ describe("Northcoders News API", () => {
...
+
+  it("Status: 404 - Should respond with a message declaring that the specified article doesn't exist on the PATCH request", () => {
+    return request(app)
+      .get("/api/articles/123456789")
+      .expect(404)
+      .then(({ text: msg }) => {
+        expect(msg).toBe("Article not found");
+      });
+  });
+
```
Comment (scar1377) [inline __tests__/app.test.js:170]: Can also add another test 400 when sending {} as the body

**Review by scar1377 — COMMENTED** [review]
code looks good. The only only I would suggest is adding another test 400 with empty object or {newVotes: 1}. Go ahead and merge to main after updating

File: __tests__/app.test.js (hunk 1 above)
Comment (clam119) [reply __tests__/app.test.js:170]: Brilliant, thank you for the feedback Scarlett, I'll go ahead and add it in and merge up to main!
